tests/Resources/ti.filesystem.file.test.js

```diff
@@ -0,0 +1,505 @@
+/*
+ * Appcelerator Titanium Mobile
+ * Copyright (c) 2011-2017 by Appcelerator, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+
+var should = require('./utilities/assertions'),
+	utilities = require('./utilities/utilities');
+
+describe('Titanium.Filesystem.File', function () {
+	it('apiName', function () {
+		var file = Ti.Filesystem.getFile('app.js');
+		should(file).have.readOnlyProperty('apiName').which.is.a.String;
+		should(file.apiName).be.eql('Ti.Filesystem.File');
+	});
+
+	// Check if name exists and returns string
+	it('name', function () {
+		var file = Ti.Filesystem.getFile('app.js');
+		should(file).have.a.readOnlyProperty('name').which.is.a.String;
+		should(file.name).be.eql('app.js');
+	});
+
+	// Check if nativePath exists and returns string
+	it('nativePath', function () {
+		var file = Ti.Filesystem.getFile('app.js');
+		should(file).have.a.readOnlyProperty('nativePath').which.is.a.String;
+	});
+
+	// Check if resolve exists and returns string
+	it('#resolve()', function () {
+		var file = Ti.Filesystem.getFile('app.js');
+		should(file.resolve).not.be.undefined;
+		should(file.resolve).be.a.Function;
+		var value = file.resolve();
+		should(value).not.be.undefined;
+		should(value).be.a.String;
+		// On Windows, it returns native path
+		if (utilities.isWindows()) {
+			should(value).be.eql(file.nativePath);
+		}
+	});
+
+	// Check if executable exists and returns boolean
+	it('executable', function () {
+		var file = Ti.Filesystem.getFile('app.js');
+		should(file).have.a.readOnlyProperty('executable').which.is.a.Boolean;
+	});
+
+	// Check if hidden exists and returns boolean
+	it('hidden', function () {
+		var file = Ti.Filesystem.getFile('app.js');
+		should(file).have.a.readOnlyProperty('hidden').which.is.a.Boolean;
+	});
+
+	// Check if readonly exists and returns boolean
+	it('readonly', function () {
+		var file = Ti.Filesystem.getFile('app.js');
+		should(file).have.a.readOnlyProperty('readonly').which.is.a.Boolean;
+	});
+
+	// Check if writable exists and returns boolean
+	it('writable', function () {
+		var file = Ti.Filesystem.getFile('app.js');
+		should(file).have.a.readOnlyProperty('writable').which.is.a.Boolean;
+	});
+
+	// Check if symbolicLink exists and returns boolean
+	it('symbolicLink', function () {
+		var file = Ti.Filesystem.getFile('app.js');
+		should(file).have.a.readOnlyProperty('symbolicLink').which.is.a.Boolean;
+	});
+
+	it('parent', function () {
+		var file = Ti.Filesystem.getFile('app.js');
+		// parent may be null if at root?
+		//should(file.parent).be.ok; // not null or undefined. should(file).not.be.null causes a stack overflow somehow.
+		should(file).have.a.readOnlyProperty('parent');
```

Comment:
I'd assume we'd also want to verify the value is another Ti.Filesystem.File. Also that at some level you can't get a parent object and get null back if you try to go up a level above the app root or whatever the directory tree sandbox happens to be.